src/components/LibraryDetailPage.tsx

```diff
@@ -65,13 +74,12 @@ export class LibraryDetailPage extends React.Component<LibraryDetailPageProps, L
     );
   }
 
-  async submit(data): Promise<void> {
+  async submit(data: FormData): Promise<void> {
     await this.props.editStages(data);
-    await this.props.fetchLibrary(this.props.uuid);
-
-    let libraryStage = this.props.fullLibrary.library_stage;
-    let registryStage = this.props.fullLibrary.registry_stage;
+    await this.props.fetchLibrary(this.props.library.uuid);
 
+    let libraryStage = this.props.fullLibrary.stages.library_stage;
+    let registryStage = this.props.fullLibrary.stages.registry_stage;
```

Comment:
You should update the interface.

Reply:
Never mind, it wasn't updated for me at first but now I see the update 👍

Reply:
What do you mean?

Reply:
The interface wasn't updated when I checked at first but later on it was fine. Weird.